src/main/java/me/earth/earthhack/impl/core/mixins/entity/living/MixinEntityLivingBase.java

```diff
@@ -429,6 +439,31 @@ public void swingArmHook(EnumHand hand, CallbackInfo ci)
         }
     }
 
+    @Inject(method = "getArmSwingAnimationEnd", at = @At("HEAD"), cancellable = true)
+    public void getArmSwingAnimationEndHook(CallbackInfoReturnable<Integer> cir) {
+        int swingSpeed = VIEW_MODEL.isEnabled() ? VIEW_MODEL.get().swingSpeed.getValue() : 6;
+        EntityPlayerSP player = Minecraft.getMinecraft().player;
+        if (player == null) {
```

Comment:
`player` is never referenced again? Also this will affect every single `EntityLivingBase`, so an instanceOf check for `EntityPlayerSP` is what I normally do.